tests/php/sync/test_class.jetpack-sync-callables.php

```diff
@@ -1053,6 +1053,50 @@ public function test_sync_callable_recursive_gets_checksum() {
 		$this->assertTrue( ! empty( $synced_value ), 'We couldn\'t synced a value!' );
 	}
 
+	/**
+	 * Test get_hosting_provider() callable to ensure that known hosts have the
+	 * right hosting provider returned.
+	 *
+	 * @return void
+	 */
+	public function test_get_hosting_provider_callable() {
+		Constants::set_constant( 'GD_SYSTEM_PLUGIN_DIR', 'set' );
+		if ( Constants::is_defined( 'GD_SYSTEM_PLUGIN_DIR' ) || class_exists( '\\WPaaS\\Plugin' ) ) {
```

Comment:
Drive-by comment: it feels a bit odd to have those checks in tests. I'd expect that the `::is_defined` check is not necessary because we just set the constant on the previous line; and instead of the `class_exists` check I'd suggest defining the class or even better - wrapping the `class_exists` check in another method that could be mocked for testing purposes.

Reply:
@tyxla thanks for the suggestion on this! It led me to make a couple of other changes to including splitting out the different types of checks and updating the tests.